src/Environment.jl

```diff
@@ -0,0 +1,42 @@
+
+const _NREFS = Ref(0)
+const _INITIALIZED = Ref(false)
+
+function Init(;p4est_verbosity_level=P4est_wrapper.SC_LP_DEFAULT,finalize_atexit=true)
+  if !MPI.Initialized()
+      MPI.Init()
+  end
+  if finalize_atexit
+    atexit(Finalize)
+  end
+  Finalize()
+
+  sc_init(parts.comm, Cint(true), Cint(true), C_NULL, p4est_verbosity_level)
+  p4est_init(C_NULL, p4est_verbosity_level)
+  _INITIALIZED[] = true
+
+  return nothing
+end
+
+function Initialized()
+  return _INITIALIZED[]
+end
+
+function Finalize()
+  if Initialized()
+    GC.gc() # Finalize all object out of scope at this point
+    if _NREFS[] != 0
+      @warn "$(_NREFS[]) objects still not finalized before calling GridapP4est.Finalize()"
+    end
+    _NREFS[] = 0
```

Comment:
Should we set `_INITIALIZED` to zero as well?

Reply:
Probably. I still have to figure out how the whole `INITIALIZED` thing works. In GridapPETSc, PETSc cannot work without initialization. However I think P4est can. So I need to test things.

Reply:
Resolved by [e9918ec](https://github.com/gridap/GridapP4est.jl/pull/27/commits/e9918ecb2a5276e422c9815c95f88e9b78bb5492)